src/scikit_build_core/settings/skbuild_read_settings.py

```diff
@@ -273,8 +273,12 @@ def __init__(
                 new_min_cmake = "3.15"
             self.settings.cmake.version = SpecifierSet(f">={new_min_cmake}")
 
+        default_cmake_minimum = "3.15"
+        if self.settings.cmake.preset:
+            default_cmake_minimum = "3.19"
```

Comment:
I don't think I am handling this part correctly. Generally it should not be tied to the `auto-cmake` version, but not sure where to feed it to override potential values set by the user. Or we just fail it at `builder.cmake`?